src/basic/socket-util.c

```diff
@@ -1472,3 +1472,71 @@ int connect_unix_path(int fd, int dir_fd, const char *path) {
 
         return RET_NERRNO(connect(fd, &sa.sa, salen));
 }
+
+int socket_address_parse_unix(SocketAddress *a, const char *s) {
+        struct sockaddr_un un;
+        int r;
+
+        assert(a);
+        assert(s);
+
+        if (!IN_SET(*s, '/', '@'))
+                return -EPROTO;
+
+        r = sockaddr_un_set_path(&un, s);
+        if (r < 0)
+                return r;
+
+        *a = (SocketAddress) {
+                .sockaddr.un = un,
+                .size = r,
+        };
+
+        return 0;
+}
+
+int socket_address_parse_vsock(SocketAddress *a, const char *s) {
+        /* AF_VSOCK socket in vsock:cid:port notation */
+        const char *cid_start = s + STRLEN("vsock:");
```

Comment:
i'd really move this down, after the startswith() check below.

static checkers don#t really like pointers that point into invalid memory, and this one might.

Reply:
or actually, rewrite the whole thing to be:

```c
cid_start = startswith(s, "vsock:");
if (!cid_start)
        return -EPROTO;
```

i..e get rid of the STRLEN(), taking benefit of the fact that startswith() returns a very useful pointer to the first character after the prefix